### ISSUE TITLE
ocamlbuild not found

### ISSUE THREAD

**Issue by Sydney-o9** [facebook/hhvm#5367]
This seems to be trickier than I thought..

````
$ sudo make

...
cat: utils/get_build_id.gen.c: No such file or directory
/bin/sh: ocamlbuild: command not found
make[4]: *** [build-hack-native-deps] Error 127
make[3]: *** [_build/hh_server.native] Error 2
make[2]: *** [hphp/hack/CMakeFiles/hack] Error 2
make[1]: *** [hphp/hack/CMakeFiles/hack.dir/all] Error 2
make: *** [all] Error 2
````

It is properly installed though... 
````
root@localhost: /tmp $ which ocamlbuild
/usr/local/bin/ocamlbuild
root@localhost: /tmp $ ocaml -version
The OCaml toplevel, version 4.01.0
````

It seems to be a path issue but why is make not picking up the path ?

**Comment by simonwelsh** [comment]
Switching to root for `sudo` is changing your path. Make sure root has `/usr/local/bin` in its path.

**Comment by jwatzman** [comment]
Yeah, ideally we'd pick up on this in the Hack makefile (have cmake detect ocamlbuild and give us the path so we can call via absolute path) but until then just run `cmake` and `make` as the same user.

**Comment by Sydney-o9** [comment]
Thanks @simonwelsh and @jwatzman. I was actually working on the documentation for people who want to [install HHVM on CentOS 6.6](https://github.com/facebook/hhvm/wiki/Building-and-installing-HHVM-on-CentOS-6.6)
The issue appears at [Install OCaml](https://github.com/facebook/hhvm/wiki/Building-and-installing-HHVM-on-CentOS-6.6#13-install-ocaml).

I think it would be awesome if anyone could have a look and give feedback on this!

**Comment by Sydney-o9** [comment]
For reference, [sudo changes PATH - why?](http://stackoverflow.com/questions/257616/sudo-changes-path-why) explains why we get that error message on CentOS.

**Comment by jwatzman** [comment]
Why are you building HHVM as root? You should be able to do all of the steps here https://github.com/facebook/hhvm/wiki/Building-and-installing-HHVM-on-CentOS-6.6#15-install-hiphop as a normal user, and then finish up with a `sudo make install`.

**Comment by lexidor** [comment]
I am going over old issues on this repository, to see which ones unlikely to be resolved.

This issue demonstrates a build error. This issue is very, very old. The CI builds don't have this problem.

If you are experiencing build errors, don't let the closing of this issue discourage you from filing a new issue.
